src/strat_engine/blockdev.rs

```diff
@@ -0,0 +1,372 @@
+        if dev_size < MIN_DEV_SIZE {
+            return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
+                                                      format!("{} too small, {} minimum",
+                                                             path.display(),
+                                                             ByteSize::b(MIN_DEV_SIZE as usize)
+                                                                 .to_string(true)))));
+        }
+
+        let mut bd = BlockDev {
+            stratis_id: stratis_id.to_owned(),
+            id: Uuid::new_v4().to_simple_string(),
+            dev: dev,
+            path: path.to_owned(),
+            sectors: Sectors(dev_size / SECTOR_SIZE),
+            mdaa: MDA {
+                last_updated: Timespec::new(0, 0),
+                used: 0,
+                length: (*mda_size / 2 * SECTOR_SIZE) as u32,
+                crc: 0,
+                offset: SectorOffset(0),
+            },
+            mdab: MDA {
+                last_updated: Timespec::new(0, 0),
+                used: 0,
+                length: (*mda_size / 2 * SECTOR_SIZE) as u32,
+                crc: 0,
+                offset: SectorOffset(*mda_size / 2),
+            },
+            mda_sectors: mda_size,
+            reserved_sectors: MDA_RESERVED_SIZE,
+        };
+
+        try!(bd.write_sigblock());
+
+        Ok(bd)
+    }
+
+    pub fn setup(path: &Path) -> EngineResult<BlockDev> {
+        let dev = try!(Device::from_str(&path.to_string_lossy()));
+
+        let mut f = try!(OpenOptions::new()
+            .read(true)
+            .open(path)
+            .map_err(|_| {
+                io::Error::new(ErrorKind::PermissionDenied,
+                               format!("Could not open {}", path.display()))
+            }));
+
+        let mut buf = [0u8; SECTOR_SIZE as usize];
+        try!(f.seek(SeekFrom::Start(SECTOR_SIZE)));
+        try!(f.read(&mut buf));
+
+        if &buf[4..20] != STRAT_MAGIC {
+            return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
+                                                      format!("{} is not a Stratis blockdev",
+                                                              path.display()))));
+        }
+
+        let crc = crc32::checksum_ieee(&buf[4..SECTOR_SIZE as usize]);
+        if crc != LittleEndian::read_u32(&buf[..4]) {
+            dbgp!("{} header CRC failed", path.display());
+            return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
+                                                      format!("{} header CRC failed",
+                                                              path.display()))));
+            // TODO: Try to read end-of-disk copy
+        }
+
+        let stratis_id = from_utf8(&buf[32..64]).unwrap();
+        let id = from_utf8(&buf[64..96]).unwrap();
+
+        let mda_size = Sectors(LittleEndian::read_u32(&buf[160..164]) as u64);
```

Comment:
See https://github.com/stratis-storage/stratisd/issues/6#issuecomment-243873443 In summary, I think we want --force for blockdevs so that people who know what they want can do it without resorting to `dd`. (That thread is really more about --force for creating filesystems, and kind of assumes we have --force for blockdevs.)